virtcontainers/sandbox.go

```diff
@@ -1546,3 +1546,26 @@ func (s *Sandbox) AppendDevice(device api.Device) error {
 	}
 	return fmt.Errorf("unsupported device type")
 }
+
+// AddDevice will add a device to sandbox
+func (s *Sandbox) AddDevice(info config.DeviceInfo) (api.Device, error) {
+	if s.devManager == nil {
+		return nil, fmt.Errorf("device manager isn't initialized")
+	}
+
+	b, err := s.devManager.NewDevice(info)
+	if err != nil {
+		return nil, err
+	}
+
+	err = s.devManager.AttachDevice(b.DeviceID(), s)
```

Comment:
```go
if err := s.devManager.AttachDevice(b.DeviceID(), s); err != nil {
```

Reply:
Oh, you're right.

I once used a global `err` for doing the rolling back, then I realized we may not need a roll back for this. 
I will change it, thanks!